heat/regression/lasso.py

```diff
@@ -116,17 +116,29 @@ def fit(self, X, y):
         # Get number of model parameters
         _, n = X.shape
 
+        assert len(y.shape) <= 2, "y.shape needs to be 1 or 2 dimensional"
+        assert len(X.shape) == 2, "X.shape needs to be 2 dimensional"
```

Comment:
should these be raises instead?

Reply:
Would also claim that should be consistent exceptions

Reply:
Is there a predefined error type for shapes in Python?

Reply:
Should be ValueError